### PR TITLE
MIQ Fix 18th Aug

### PR DESCRIPTION
**Pull request by monagjr**

### CODE REVIEW

**Comment by darianbird** [comment]
Thanks for putting this together. It worked for me last night (Europe time) and stopped on the same slots as the twitter bots. It didn't select the dates or beep (I tried deleting date.elem.click(); to test beep() but no luck there) though - does it do that for you? It also seemed to ignore my dates and stopped on anything available. It's possible I messed something up copying the code over.

**Comment by monagjr** [comment]
Thanks @bennyggg for test and reporting.
I've added one more update to the code to match the new MIQ update, and this will make it works as expected (i.e clicking the date, beep, match target dates only)

**Comment by ricoboirn** [comment]
Hi @monagjr  trying your new code with testDates. I couldn't get it to work. I have no idea to coding, but kindly guide me if I'm doing something wrong. Thanks.

// Set to true to turn on test mode
let testDates = true;


if (await findAvailability(page)) {
        const status = "AVAILABLE! Found at: " + new Date().toLocaleString();
        console.log(status)
        updateElectronStatus('status-count', status);
        testDates = ['2021-09-26'];

**Comment by monagjr** [comment]
Thanks @ricoboirn  for testing and reporting.
I've updated the code part related to "test mode", it now works as expected.

**Comment by darianbird** [comment]
> Thanks @bennyggg for test and reporting.
> I've added one more update to the code to match the new MIQ update, and this will make it works as expected (i.e clicking the date, beep, match target dates only)

Thanks. I've updated to your newest version and it works with testDates

**Comment by ricoboirn** [comment]
@monagjr  Awesome, working as intended now. I'm on my 4th week now trying my luck securing for a friend. Thanks.

**Comment by monagjr** [comment]
Thanks @ricoboirn @bennyggg for confirmation
I hope this will help more to secure their needed spots

**Comment by monagjr** [comment]
@jvolker @alexDrinkwater @stilljake can you merge this with base code and create new release binaries?

**Comment by Jefsmango** [comment]
Hi guys, sorry to butt in, thanks much for doing this, this is a lot of help for non-programmers such as myself. If I may ask a favor? would you be so kind to update the MIQ-Booking-Assistance v0.10.3 exe file so that we can redownload and use it as well? Really need to book a slot for my daughter. Thanks a lot

**Comment by rufdev** [comment]
no beep again.. sad..

**Comment by stilljake** [comment]
> @jvolker @alexDrinkwater @stilljake can you merge this with base code and create new release binaries?

Sorry, I don't have write access!

**Comment by monagjr** [comment]
@Jefsmango and anyone else who needs ready-made pre-built version of the APP latest version (v0.10.4):

https://github.com/monagjr/miq-nz-booking-tools/releases/latest

-----------------
https://www.buymeacoffee.com/mojr

File: MIQ-Booking-Assistance/miq-assistance.js [hunk 1]
```diff
@@ -142,12 +148,17 @@ async function findAvailability(page) {
     return await page.evaluate((myDates, testDates, findAnyDate, checkedCount) => {
         const nextElem = document.getElementById('form_next');
         nextElem.scrollIntoView();
-
-        const availableDateElements = [...document.querySelectorAll('[class$="_item"] :not(.no)')].filter(d => d.ariaLabel);
+        
+        const prefix = document.querySelector('.mo-6 > div').getAttribute('class')
```
Comment (sebjamesdev) [inline MIQ-Booking-Assistance/miq-assistance.js:152]: I don't think it's guaranteed that the class will be `mo-6` I've seen other values here.

Comment (monagjr) [reply MIQ-Booking-Assistance/miq-assistance.js:152]: @sebjamesdev are you sure there are other values for the class than `mo-6` ? what other values have you found?
 this is by the way a class for the calendar container.

Comment (sebjamesdev) [reply MIQ-Booking-Assistance/miq-assistance.js:152]: Sorry, my bad that was the logged out portal calendar.

**Comment by Jefsmango** [comment]
Very much appreciated @monagjr , they must have found a way to mess with the script output, it works ok but after a few tries, it stops on the date but not selecting it and no more beeping. and even when you click on the date you cant fill up the recaptcha because it doesnt stop refreshing...

**Comment by monagjr** [comment]
@Jefsmango thanks for reporting this, I'll check it if something from code, or temp issue on your end.

Can you please open a discussion topic under "bugs / issues" category here?

https://github.com/monagjr/miq-nz-booking-tools/discussions/new

**Comment by jvolker** [comment]
Thanks, @monagjr. I've tried your changes and was able to get an alert. I've just merged this. New releases will follow in a few minutes.

**Comment by monagjr** [comment]
Welcome back @jvolker :)

**Pull request closed without merging**